### PR TITLE
 driver/audio: update i2s timeout value

### PR DESCRIPTION
**Pull request by ritesh55555**
This commit updates the value of timeout for i2s transfer time. Earlier it was fixed but now  it is defined respect to buffer size, sample rate, and bitrate.
 i2s transfer time = (buffer size * num of buffer) / (sample rate * channel * bit per sample)
 timeout = i2s transfer time + some offset

this commit also enable config_amebasmart_i2c2 in audio defconfig, so that speaker will work properly with PM operation.

### CODE REVIEW

File: os/drivers/audio/syu645b.c [hunk 1]
```diff
@@ -59,6 +57,11 @@
 #define CONFIG_SYU645B_NUM_BUFFERS       4
 #endif
 
+/* Converting buffer size from bytes to bits */
+#define SYU645B_I2S_TRANSFER_TIME_MS     (CONFIG_SYU645B_BUFFER_SIZE * CONFIG_SYU645B_NUM_BUFFERS * 8 * 1000) / (SYU645B_SAMPLE_FREQUENCY_IN_HZ * SYU645B_CHANNELS * SYU645B_BPSAMP)
```
Comment (abhishek-samsung) [inline os/drivers/audio/syu645b.c:61]: Ideally we should be able to change the sampling rate and channels. So, we should not use hardcoded values. I think instead we can use,

```
                priv->samprate = caps->ac_controls.hw[0];
                priv->nchannels = caps->ac_channels;
                priv->bpsamp = caps->ac_controls.b[2];
```

File: build/configs/rtl8730e/audio/defconfig [hunk 2]
```diff
@@ -164,7 +164,7 @@ CONFIG_AMEBASMART_SPI_EXCHANGE=y
 CONFIG_AMEBASMART_I2C=y
 CONFIG_AMEBASMART_I2C0=y
 # CONFIG_AMEBASMART_I2C1 is not set
-# CONFIG_AMEBASMART_I2C2 is not set
+CONFIG_AMEBASMART_I2C2=y
```
Comment (kishore-sn) [inline build/configs/rtl8730e/audio/defconfig:167]: Please make a separate commit for this. It is not related to i2s timeout.

File: os/drivers/audio/alc1019.c [hunk 3]
```diff
@@ -745,8 +744,13 @@ static int alc1019_enqueuebuffer(FAR struct audio_lowerhalf_s *dev, FAR struct a
 		alc1019_givesem(&priv->devsem);
 		return OK;
 	}
+
+	/* Converting buffer size from bytes to bits 
+	 * and then calculating the i2s transfer time 
+	 * and adding 10 ms as an offset for timeout */
+	timeout = (CONFIG_SYU645B_BUFFER_SIZE * CONFIG_SYU645B_NUM_BUFFERS * 8 * 1000) / (priv->samprate * priv->nchannels * priv->bpsamp) + 10;
```
Comment (kishore-sn) [inline os/drivers/audio/alc1019.c:751]: Let us not use hardcode value of 10. Let us name it as MIN_I2S_TIMEOUT_MS.

File: os/drivers/audio/syu645b.c [hunk 4]
```diff
@@ -446,6 +444,21 @@ static int syu645b_configure(FAR struct audio_lowerhalf_s *dev, FAR const struct
...
+			syu645b_exec_i2c_script(priv, codec_set_samprate_32k_script, sizeof(codec_set_samprate_32k_script) / sizeof(t_codec_init_script_entry));
+		} else if (priv->samprate == AUDIO_SAMP_RATE_44K) {
+			syu645b_exec_i2c_script(priv, codec_set_samprate_44k_script, sizeof(codec_set_samprate_44k_script) / sizeof(t_codec_init_script_entry));
+		} else if (priv->samprate == AUDIO_SAMP_RATE_48K) {
+			syu645b_exec_i2c_script(priv, codec_set_samprate_48k_script, sizeof(codec_set_samprate_48k_script) / sizeof(t_codec_init_script_entry));
+		} else if (priv->samprate == AUDIO_SAMP_RATE_96K) {
+			syu645b_exec_i2c_script(priv, codec_set_samprate_96k_script, sizeof(codec_set_samprate_96k_script) / sizeof(t_codec_init_script_entry));
+		} else {
+			auddbg("ERROR: Unsupported sample rate: %d\n", priv->samprate);
+		}
```
Comment (kishore-sn) [inline os/drivers/audio/syu645b.c:460]: You have not mentioned this change in the commit description. Please add.

File: os/drivers/audio/syu645b.c [hunk 5]
```diff
@@ -736,8 +752,13 @@ static int syu645b_enqueuebuffer(FAR struct audio_lowerhalf_s *dev, FAR struct a
 		syu645b_givesem(&priv->devsem);
 		return OK;
 	}
-	
-	ret = I2S_SEND(priv->i2s, apb, syu645b_txcallback, priv, SYU645B_I2S_TIMEOUT_MS);
+
+	/* Converting buffer size from bytes to bits 
+	 * and then calculating the i2s transfer time 
+	 * and adding 10 ms as an offset for timeout */
+	timeout = (CONFIG_SYU645B_BUFFER_SIZE * CONFIG_SYU645B_NUM_BUFFERS * 8 * 1000) / (priv->samprate * priv->nchannels * priv->bpsamp) + SYU645B_MIN_I2S_TIMEOUT_MS;
```
Comment (kishore-sn) [inline os/drivers/audio/syu645b.c:759]: Can you use some declaration instead of using hard coded values?

File: build/configs/rtl8730e/audio/defconfig (hunk 2 above)
Comment (kishore-sn) [reply build/configs/rtl8730e/audio/defconfig]: Can you pls check this comment?

**Comment by Taejun-Kwon** [comment]
How bixby works with this PR?

File: os/drivers/audio/alc1019.c [hunk 6]
```diff
@@ -44,7 +44,9 @@
 #include "alc1019.h"
 #include "alc1019scripts.h"
 
-#define ALC1019_I2S_TIMEOUT_MS 200
+#define ALC1019_MIN_I2S_TIMEOUT_MS  10
+#define BYTE_TO_BIT_FACTOR          8
+#define SEC_TO_MSEC_FACTOR	    1000
```
Comment (kishore-sn) [inline os/drivers/audio/alc1019.c:49]: Can we move these declarations to some common header file?
MIN_I2S_TIMEOUT_MS must be common for all I2S?
And the other two declarations are very common. I think its already present somewhere in some header files.

Comment (ritesh55555) [reply os/drivers/audio/alc1019.c:49]: MIN_I2S_TIMEOUT_MS is a configurable parameter , but we can keep it common for all I2S . I will add these in header file

**Review by kishore-sn — APPROVED** [review]

File: os/drivers/audio/alc1019.c [hunk 7]
```diff
@@ -745,8 +747,13 @@ static int alc1019_enqueuebuffer(FAR struct audio_lowerhalf_s *dev, FAR struct a
 		alc1019_givesem(&priv->devsem);
 		return OK;
 	}
+
+	/* Converting buffer size from bytes to bits 
+	 * and then calculating the i2s transfer time 
+	 * and adding 10 ms as an offset for timeout */
+	timeout = (CONFIG_ALC1019_BUFFER_SIZE * CONFIG_ALC1019_NUM_BUFFERS * BYTE_TO_BIT_FACTOR * SEC_TO_MSEC_FACTOR) / (priv->samprate * priv->nchannels * priv->bpsamp) + I2S_TIMEOUT_OFFSET_MS;
```
Comment (r-prabu) [inline os/drivers/audio/alc1019.c:754]: check the code indentation is applied properly.

File: os/drivers/audio/alc1019.c [hunk 8]
```diff
@@ -44,7 +44,8 @@
 #include "alc1019.h"
 #include "alc1019scripts.h"
 
-#define ALC1019_I2S_TIMEOUT_MS 200
+#define BYTE_TO_BIT_FACTOR          8
```
Comment (gSahitya-samsung) [inline os/drivers/audio/alc1019.c:47]: Changes look fine

**Review by gSahitya-samsung — COMMENTED** [review]
Changes look f

**Review by neel-samsung — APPROVED** [review]
update i2s timeout PR looks good

**Comment by ritesh55555** [comment]
> How bixby works with this PR?

bixby works fine with this PR, tested on public and separate build product code
